shell/src/main/java/alluxio/cli/validation/hdfs/HdfsProxyUserValidationTask.java

```diff
@@ -0,0 +1,106 @@
+package alluxio.cli.validation.hdfs;
+
+import alluxio.cli.ValidationUtils;
+import alluxio.cli.validation.ApplicableUfsType;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.status.UnauthenticatedException;
+import alluxio.security.authentication.AuthType;
+import alluxio.security.authentication.ImpersonationAuthenticator;
+import alluxio.security.user.UserState;
+
+import java.util.Map;
+
+/**
+ * Validates the proxyuser configurations for alluxio in hdfs.
+ * */
+@ApplicableUfsType(ApplicableUfsType.Type.HDFS)
+public class HdfsProxyUserValidationTask extends HdfsConfValidationTask {
+  /**
+   * Creates a new instance of {@link HdfsProxyUserValidationTask}
+   * for validating proxyuser configuration.
+   *
+   * @param path the UFS path
+   * @param conf the UFS configuration
+   */
+  public HdfsProxyUserValidationTask(String path, AlluxioConfiguration conf) {
+    super(path, conf);
+  }
+
+  @Override
+  public String getName() {
+    return "ValidateProxyUserConf";
+  }
+
+  private String getCurrentUser() throws UnauthenticatedException {
+    UserState s = UserState.Factory.create(mConf);
+    return s.getUser().getName();
+  }
+
+  private ValidationUtils.TaskResult validateProxyUsers(String userName) {
+    String proxyUserKey = String.format("hadoop.proxyuser.%s.users", userName);
+    String proxyGroupKey = String.format("hadoop.proxyuser.%s.groups", userName);
+    String proxyUsers = mCoreConf.getOrDefault(proxyUserKey, "");
+    String proxyGroups = mCoreConf.getOrDefault(proxyGroupKey, "");
+
+    // Neither proxy users or groups is configured in core-site.xml
+    if (proxyUsers.equals("") && proxyGroups.equals("")) {
+      mMsg.append(String.format("Alluxio is running as user %s. But neither %s or %s is "
+              + "configured in hadoop configuration. Alluxio is not able to perform "
+              + "impersonation.%n", userName, proxyUserKey, proxyGroupKey));
+      mAdvice.append(String.format("Please enable Alluxio user %s to impersonate.%n", userName));
+
+      return new ValidationUtils.TaskResult(ValidationUtils.State.FAILED, getName(), mMsg.toString(),
+              mAdvice.toString());
+    }
+
+    // If proxy users/groups is *, alluxio can impersonate anyone
+    if (proxyUsers.equals(ImpersonationAuthenticator.WILDCARD)
+            || proxyGroups.equals(ImpersonationAuthenticator.WILDCARD)) {
+      mMsg.append(String.format("Alluxio user %s can impersonate as any user/group in HDFS.%n",
+              userName));
+      return new ValidationUtils.TaskResult(ValidationUtils.State.OK, getName(),
+              mMsg.toString(), mAdvice.toString());
+    }
+
+    // There are proxyable users and groups for the Alluxio user in HDFS,
+    // but we cannot know if that is a full set.
+    // Leave a warning for the user to double check.
+    // TODO(jiacheng): can we do better check than this?
```

Comment:
@madanadit PTAL at the impersonation check now, updated per our discussion today. Now the logic goes as follows:

```
// Skipped if authType is NOSASL
validateProxyUsers(String alluxioUser) {
// if neither hadoop.proxyuser.alluxioUser.users and hadoop.proxyuser.alluxioUser.groups are configured -> ERROR
// If either is *, OK
// Otherwise  -> WARN and let the user check. 
}
```

What do you think about the final WARN scenario? Maybe hadoop.proxyuser.alluxioUser.users or hadoop.proxyuser.alluxioUser.groups has some config but we don't really know if that's what we want. I left a TODO in case we want to add a little more to this part.